contrib/systemd/xdg-desktop-portal-wlr.service.in

```diff
@@ -0,0 +1,7 @@
+[Unit]
+Description=Portal service (wlroots implementation)
```

Comment:
We probably want to depend on `graphical-session.target`, among other things. See:

https://github.com/emersion/mako/blob/master/contrib/systemd/mako.service.in

Reply:
That's... complicated.

One of the things to consider is that `graphical-session.target` is disabled in default preset and not started by sway. Any hard dependency would prevent service from starting and soft dependency (PartOf/After) would be a noop.
`sway-session.target` from sway wiki should address that, but it is not a part of sway distribution and thus ignored by downstreams.

Another thing is that none of xdg-*.service files I saw contains dependencies. If I understand correctly it's because they are meant to be purely dbus activated and excluded from systemd dependency management.

Reply:
Okay. Hmm. Why do we want a systemd service file then? Isn't a D-Bus service file enough?

If we do want that service file, is there a way to prevent the service from being enabled?

Reply:
>One of the things to consider is that graphical-session.target is disabled in default preset and not started by sway.

But graphical-session.target should be managed by… the display manager? Not by Sway itself?

Reply:
> Okay. Hmm. Why do we want a systemd service file then? Isn't a D-Bus service file enough?

For all the systemd voodoo magic like session management, crash handling and per-user configuration.
For instance, I have an user override for system package configs that redefines `ExecStart=` to use the arguments I need. That's not possible with system-installed D-Bus service.

> If we do want that service file, is there a way to prevent the service from being enabled?

It already omits `[Install]`  and `WantedBy=` which is sufficient.

> But graphical-session.target should be managed by… the display manager? Not by Sway itself?

Not according to my testing and knowledge. graphical-session.target is usually bound to $softwarename-session and $softwarename-session should be started by $software itself. For example, GNOME starts the target from gnome-session executable.

Reply:
I see, thanks for the explanations